### PR TITLE
[Plugin] Add support for SetCheatAction and cheat retrieval

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

File: src/openrct2/scripting/ScCheats.hpp [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+#pragma once
+
+#ifdef ENABLE_SCRIPTING
+
+#    include "../Cheats.h"
+#    include "Duktape.hpp"
+#    include "ScriptEngine.h"
+
+namespace OpenRCT2::Scripting
```
Comment (tupaschoal) [inline src/openrct2/scripting/ScCheats.hpp:18]: Why not?
```suggestion
using OpenRCT2::Scripting;
```

Then you won't have to open a huge namespace and with the extra 4 spaces maybe somethings won't break in as many lines hahaha

Comment (IntelOrca) [reply src/openrct2/scripting/ScCheats.hpp:18]: You shouldn't do this in header files. Unless scoped which this is not.

Comment (tupaschoal) [reply src/openrct2/scripting/ScCheats.hpp:18]: Ah, I hadn't noticed it was a header

**Comment by tupaschoal** [comment]
There is one of the CI checks failing

File: src/openrct2/scripting/ScriptEngine.cpp [hunk 2]
```diff
@@ -369,7 +369,7 @@ ScriptEngine::ScriptEngine(InteractiveConsole& console, IPlatformEnvironment& en
 void ScriptEngine::Initialise()
 {
     auto ctx = (duk_context*)_context;
-    ScCheats::Register(ctx);
+    ScCheats::Register();
```
Comment (Broxzier) [inline src/openrct2/scripting/ScriptEngine.cpp:372]: The `Register` function still needs it, right? Just not the constructor.

**Pull request merged**